### DISCUSSION TITLE
Separate message after reaching 2000 characters

### DISCUSSION THREAD

**Discussion by Cvar1984** [discord-php/DiscordPHP#1184] [Q&A] [answered]
how do i separate reply message after reaching 2000 characters and also able to send markup message with html
```php
    $discord->listenCommand('list_chapter', function (Interaction $interaction) {
        $chapterList = Quran::listChapter();
        $list = '';
        foreach($chapterList as $chapter) {
            $list .= "{$chapter['nomor']}.{$chapter['nama']} - {$chapter['arti']}\n";
        }
        $interaction->respondWithMessage(MessageBuilder::new()->setContent($list));
    });
```
my listen command looks like this the $list variable contain a lot of character with html markup.
when i tried use `$interaction->respondWithMessage(MessageBuilder::new()->setContent());` in a loop it looks like i can't use it more than once so thats why i concatenate `$chapter` as a string in one variable called `$list` but its exceeded 2000 chars

**Comment by CommandString** [comment]
You can use [str_split](https://www.php.net/manual/en/function.str-split.php) and then either...
Send a follow up message or send a message with the channel provided by the interaction.

**Reply by Cvar1984** [reply]
it works, how about html formated message

**Reply by CommandString** [reply]
> it works, how about html formated message

Not sure if I understand completely

**Reply by Cvar1984** [reply]
my message contain standard html markup like markdown does like making text bold and italic "<b><i>" but its not getting formatted on MessageBuilder

**Reply by SQKo** [reply]
You cannot use HTML codes here, you can use markdown formatting from here: https://support.discord.com/hc/en-us/articles/210298617-Markdown-Text-101-Chat-Formatting-Bold-Italic-Underline-

**Comment by SQKo** [comment, marked as answer]
Message contents are indeed limited to 2000 characters, but if you also use embed, you can put longer description, However it is limited up to 6000 characters for every embed contents

Or you can split responses message like @CommandString said, but I suggest avoid using channel send message method.

On design notes though, Quran chapters lists are big to display in one page, I suggest you create a paginator (that, having next/previous button), but if the goal is to find & search then select a Quran chapter, i suggest using Interaction command autocomplete, the following is example (Code requires v10/dev-master branch):
```php
$discord->listenCommand(['cmd', 'sub', 'int'], function (Interaction $interaction) {
  // .. this is interaction action code, when the command is submitted
}, function (Interaction $interaction) use ($discord) {
  // .. this is auto complete suggestions, when the option is focused
  return [
    new Choice($discord, ['name' => 'one', 'value' => 1]),
    new Choice($discord, ['name' => 'two', 'value' => 2]),
    // ... up to 25
  ];
});
```
![205528837-34b857d9-cecf-45f7-afd6-047039448b36](https://github.com/discord-php/DiscordPHP/assets/87897282/c824e96e-4df3-42f6-a6ba-3bb21eef2c42)


You need to alter your command and use the builder to set the command Option `->autoComplete(true)` though

**Reply by Cvar1984** [reply]
thanks that is very informative

**Reply by CommandString** [reply]
Autocomplete, if SQKo is correct about your intented UX, is easily the best the way to go.

**Reply by CommandString** [reply]
~~also why can't I make this the answer~~

**Reply by SQKo** [reply]
You should let the OP mark the answer that worked for them, it's unethical to do without their consent, unless it's an official statement/fact from us rather than opinionated answer 😛
